### PR TITLE
test: Fix sign for expected values

### PR DESCRIPTION
**Pull request by kallewoof**
A number of `BOOST_CHECK_EQUAL` calls would result in warnings about signs.

This PR fixes signedness for all expectation values, sometimes resulting in `int` → `unsigned int`. No other code changes besides adding/removing `U` to/from values.

Running `make &> make_output_...` on master versus on this PR:
```
$ wc make_output_*
    1464    5925   90357 make_output_master
     613    1469   28370 make_output_signfixed
```
More than halves the output lines from compiling.

### CODE REVIEW

**Comment by practicalswift** [comment]
utACK 08f99ecf30ff994e57045ed633c0ab4e61f01ea6

**Comment by MarcoFalke** [comment]
What tool have you used to see the warnings?

**Comment by kallewoof** [comment]
@MarcoFalke macOS clang (g++)

**Comment by kallewoof** [comment]
Running `make &> make_output_...` on master versus on this PR:
```
$ wc make_output_*
    1464    5925   90357 make_output_master
     613    1469   28370 make_output_signfixed
```
More than halves the output lines from compiling.

File: src/test/coins_tests.cpp [hunk 1]
```diff
@@ -480,16 +480,16 @@ BOOST_AUTO_TEST_CASE(ccoins_serialization)
     Coin cc1;
     ss1 >> cc1;
     BOOST_CHECK_EQUAL(cc1.fCoinBase, false);
-    BOOST_CHECK_EQUAL(cc1.nHeight, 203998);
-    BOOST_CHECK_EQUAL(cc1.out.nValue, 60000000000ULL);
+    BOOST_CHECK_EQUAL(cc1.nHeight, 203998U);
+    BOOST_CHECK_EQUAL(cc1.out.nValue, 60000000000LL);
```
Comment (MarcoFalke) [inline src/test/coins_tests.cpp]: nit: I think `CAmount{60000000000}` works as well

File: src/test/main_tests.cpp [hunk 2]
```diff
@@ -55,7 +55,7 @@ BOOST_AUTO_TEST_CASE(subsidy_limit_test)
         nSum += nSubsidy * 1000;
         BOOST_CHECK(MoneyRange(nSum));
     }
-    BOOST_CHECK_EQUAL(nSum, 2099999997690000ULL);
+    BOOST_CHECK_EQUAL(nSum, 2099999997690000LL);
```
Comment (MarcoFalke) [inline src/test/main_tests.cpp]: nit: Same here `CAmount{2099999997690000}`

**Comment by MarcoFalke** [comment]
utACK 08f99ecf30ff994e57045ed633c0ab4e61f01ea6

**Comment by kallewoof** [comment]
Switched to `CAmount{}` according to @MarcoFalke nits.

**Comment by jonasschnelli** [comment]
Tested ACK c55aa4f27d4fcb1d179fb1b17034476a5918ea85
(fixed my new boost::test warnings on macOS 10.12)

**Comment by MarcoFalke** [comment]
re-utACK c55aa4f27d

**Pull request merged**